openstack/canonical/etc/maas/preseeds/generic

```diff
@@ -0,0 +1,117 @@
+{{inherit "preseed_master"}}
+
+{{def proxy}}
+d-i     mirror/country string manual
+{{if node.split_arch()[0] in {'i386', 'amd64'} }}
+d-i     mirror/http/hostname string {{main_archive_hostname}}
+d-i     mirror/http/directory string {{main_archive_directory}}
+{{else}}
+d-i     mirror/http/hostname string {{ports_archive_hostname}}
+d-i     mirror/http/directory string {{ports_archive_directory}}
+{{endif}}
+{{if http_proxy }}
+d-i     mirror/http/proxy string {{http_proxy}}
+{{else}}
+d-i     mirror/http/proxy string http://{{server_host}}:8000/
+{{endif}}
+{{enddef}}
+
+{{def client_packages}}
+{{if third_party_drivers and driver}}
+d-i     pkgsel/include string cloud-init openssh-server python-software-properties vim avahi-daemon ifenslave-2.6 server^ {{driver_package}}
+{{else}}
+d-i     pkgsel/include string cloud-init openssh-server python-software-properties vim avahi-daemon ifenslave-2.6 server^
+{{endif}}
+{{enddef}}
+
+{{def driver_preseed_data}}
+{{if third_party_drivers and driver}}
+# Third party driver support.
+d-i preseed/early_command string \
+    echo -en '{{''.join(['\\x%x' % x for x in map(ord, str(install_udeb))])}}' > /tmp/install_udeb.sh && \
+    chmod +x /tmp/install_udeb.sh && \
+    /tmp/install_udeb.sh
+
+d-i apt-setup/local0/repository string deb {{driver['repository']}} {{node.get_distro_series()}} main
+d-i apt-setup/local0/comment string {{driver['comment']}}
+d-i apt-setup/local0/key string file:///tmp/maas-{{driver['package']}}/repo_key.gpg
+{{endif}}
+{{enddef}}
+
+{{def preseed}}
+{{preseed_data}}
+{{driver_preseed_data}}
+{{enddef}}
+
+{{def post_scripts}}
+# Executes late command and disables PXE.
+d-i     preseed/late_command string true && \
+    in-target sh -c 'f=$1; shift; echo $0 > $f && chmod 0440 $f $*' 'ubuntu ALL=(ALL) NOPASSWD: ALL' /etc/sudoers.d/maas && \
+    in-target wget --no-proxy "{{node_disable_pxe_url|escape.shell}}" --post-data "{{node_disable_pxe_data|escape.shell}}" -O /dev/null && \
+    in-target wget --no-proxy -O /usr/local/bin/bond-setup.sh http://10.4.4.131/MAAS/static/bsn/bond-setup.sh && \
```

Comment:
the only change to preseed is line 21, 23, 51, 52, 53.
```post_scripts``` section is executed as a last step before machine restarts and ```bond-setup.sh``` is executed on ```udevtrigger``` after all interfaces are registered and named.